Substack.js

```diff
@@ -0,0 +1,164 @@
+{
+	"translatorID": "ac3b958f-0581-4117-bebc-44af3b876545",
+	"label": "Substack",
+	"creator": "Abe Jellinek",
+	"target": "^https://[^.]+\\.substack\\.com/(p/|archive)",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2021-07-20 02:11:47"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2021 Abe Jellinek
+	
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+
+function detectWeb(doc, url) {
+	if (url.includes('/p/')) {
+		return "blogPost";
+	}
+	else if (getSearchResults(doc, true)) {
+		return "multiple";
+	}
+	return false;
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('a.post-preview-title[href*="/p/"]');
+	for (let row of rows) {
+		let href = row.href;
+		let title = ZU.trimInternal(row.textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (items) ZU.processDocuments(Object.keys(items), scrape);
+		});
+	}
+	else {
+		scrape(doc, url);
+	}
+}
+
+function scrape(doc, _url) {
+	let item = new Zotero.Item('blogPost');
+	let json = JSON.parse(text(doc, 'script[type="application/ld+json"]'));
+	
+	item.title = json.headline;
+	item.abstractNote = json.description;
+	item.date = ZU.strToISO(json.dateModified || json.datePublished);
+	item.creators.push(ZU.cleanAuthor(json.author.name, 'author'));
+	item.blogTitle = json.publisher.name;
+	item.url = attr(doc, 'link[rel="canonical"]', 'href');
+	item.attachments.push({
+		title: 'Snapshot',
+		document: doc
+	});
+	item.websiteType = 'Substack newsletter';
```

Comment:
Should this just be "Newsletter"? "Substack" seems a bit redundant here with the URL.

(I've never been clear on how exactly the type field is used by citation styles. @adam3smith, @bwiernik?)

Reply:
It's a description of the type of resource like "tweet" or "Infographic" or "Facebook page". See e.g. this sample entry by @bwiernik for APA:
https://www.zotero.org/groups/2205533/items/7AYFWZNP
So I guess the question is if substack is generic like Twitter and Facebook by now. I'd be inclined to say it is, so to keep Substack, in particular because it's nowhere else in the citation info (other than the URL).

Related question is if the item type should be blog or webpage. The two are obviously closely related and I think substack has basically replaced blog, but at least for Chicago and APA style, we mainly use webpage for "all sorts of online things"

Reply:
Yes, since the Substack name is taking up the Website Title field, let's keep "Substack" in the type field. I agree that "Substack newsletter" is similar at this point to "Facebook post" or "Tweet". 

I think making this a Blog post makes sense considering that these are serialized. I suggest we do that.

Reply:
OK, I'll leave it as `blogPost` with "Substack" in the type field. I would definitely argue that "Substack" is generic and conveys a particular type of publication that there isn't a better adjective for: an email newsletter that's also published online, often behind a paid subscription.